adapter-base/src/main/java/org/eclipse/hono/adapter/MultiMessagingSenders.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2021 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.eclipse.hono.adapter;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import org.eclipse.hono.util.TenantConstants;
+import org.eclipse.hono.util.TenantObject;
+
+import io.vertx.core.json.JsonObject;
+
+/**
+ * Encapsulates the Kafka-based and the AMQP-based sender for a message type.
+ *
+ * @param <T> The sender type (intended to be one of {@link org.eclipse.hono.adapter.client.telemetry.EventSender},
+ *            {@link org.eclipse.hono.adapter.client.telemetry.TelemetrySender}, or
+ *            {@link org.eclipse.hono.adapter.client.command.CommandResponseSender}).
```

Comment:
these particular types seem to be unrelated to this class, thus I'd rather not introduce a dependency by means of this JavaDoc comment.